### PR TITLE
td 1.2 (new formula)

### PR DESCRIPTION
**Pull request by Swatto**
Your todo list in your terminal: small and fast

### CODE REVIEW

File: Library/Formula/td.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    mkdir_p buildpath/"src/github.com/Swatto/"
+    ln_sf buildpath, buildpath/"src/github.com/Swatto/td"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "."
+    bin.install "td-1.2.0" => "td"
+  end
+
+  test do
+    system "#{bin}/td"
```
Comment (tdsmith) [inline Library/Formula/td.rb]: Can this test be expanded to exercise some function of `td`? `$HOME` is set to a temporary directory so the test can use the home directory without interfering with a user's data.

File: Library/Formula/td.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    Now you need to add "TODO_DB_PATH" to your environment variable.
+    This will be use as a fallback if your directories does not contains any ".todos" files.
+
+    Find more information here:
+    #{homepage}
+    EOS
```
Comment (tdsmith) [inline Library/Formula/td.rb]: Please remove the caveats since they aren't Homebrew-specific.

File: Library/Formula/td.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+require "language/go"
+
+class Td < Formula
+  homepage "https://github.com/Swatto/td"
+  revision 1
```
Comment (tdsmith) [inline Library/Formula/td.rb]: The `revision` line can be deleted.

File: Library/Formula/td.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  stable do
+    url "https://github.com/Swatto/td/archive/1.2.0.tar.gz"
+    sha256 "86a84cb183e2b3700a23cc5570b65ea8d133e51c5c59e321929ace727253d55b"
+  end
```
Comment (tdsmith) [inline Library/Formula/td.rb]: This doesn't need to be in a `stable` block; these can just be at the class level.

File: Library/Formula/td.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/td.rb]: If this is a new formula, it doesn't have a `revision`. The `revision` is an unwritten 0.

File: Library/Formula/td.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require "language/go"
+
+class Td < Formula
+  homepage "https://github.com/Swatto/td"
+  revision 1
+
+  stable do
```
Comment (DomT4) [inline Library/Formula/td.rb]: Probably doesn't need to be nested inside a `stable do` block unless stable has patches, or special deps/resources.

File: Library/Formula/td.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    ENV["GOBIN"] = bin
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/Swatto/"
+    ln_sf buildpath, buildpath/"src/github.com/Swatto/td"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "."
+    bin.install "td-1.2.0" => "td"
```
Comment (DomT4) [inline Library/Formula/td.rb]: To save others having to rewrite this with each bump, `bin.install "td-#{version}" => "td"` may be nicer.

File: Library/Formula/td.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/td.rb]: Oops, Sorry. Tim's comments didn't exist before I commented, they appeared once I pressed the refresh key. Sorry Tim :sweat_smile:

File: Library/Formula/td.rb (hunk 2 above)
Comment (Swatto) [inline Library/Formula/td.rb]: Since the user needs to do an operation, what should I use?

**Comment by DomT4** [comment]
Please address Tim's comments, Thanks!

**Pull request closed without merging**